snyk/client.py

```diff
@@ -30,35 +34,86 @@ def __init__(
         }
         self.api_post_headers = self.api_headers
         self.api_post_headers["Content-Type"] = "application/json"
+        self.tries = tries
+        self.backoff = backoff
+        self.delay = delay
+
+    def make_http_call(
+        self, method, url: str, headers: object, json={},
+    ) -> requests.Response:
+        resp = method(url, json=json, headers=headers,)
+        if not resp or resp.status_code >= 500:
+            raise SnykHTTPError(resp)
+        return resp
 
     def post(self, path: str, body: Any) -> requests.Response:
         url = "%s/%s" % (self.api_url, path)
         logger.debug("POST: %s" % url)
-        resp = requests.post(url, json=body, headers=self.api_post_headers)
+        resp = retry_call(
+            self.make_http_call,
+            fargs=[requests.post, url],
+            fkwargs={"json": body, "headers": self.api_post_headers},
+            tries=self.tries,
+            delay=self.delay,
+            max_delay=None,
+            backoff=self.backoff,
+            jitter=0,
+            logger=logger,
+        )
         if not resp:
             raise SnykHTTPError(resp)
         return resp
 
     def put(self, path: str, body: Any) -> requests.Response:
         url = "%s/%s" % (self.api_url, path)
         logger.debug("PUT: %s" % url)
-        resp = requests.put(url, json=body, headers=self.api_post_headers)
+        resp = retry_call(
```

Comment:
Since your using named parameters in calling `retry_call`, there is no need to specify default parameters that cannot be changed through `self`. For the sake of correctness, would you be able to remove `max_delay` and `jitter` from the function call?